### PR TITLE
Love: Pimp filter card

### PR DESCRIPTION
**Pull request by c727**
fix: https://github.com/home-assistant/ui-schema/issues/16

```yaml
    cards:
      - type: entity-filter
        filter:
          - domain: input_boolean
            state: 'on'
          - domain: light
            state: 'off'
          - entity_id: "*humid*"
        title: The new filter
        card: picture-glance
        image: https://images.crateandbarrel.com/is/image/Crate/DSC_RUGS_201804_RugsbySze?wid=1680&qlt=75&
      - type: entity-filter
        filter:
          - domain: input_boolean
            state: 'on'
          - domain: light
            state: 'off'
          - entity_id: "*humid*"
        title: The new filter
```
- supports multiple filters
- entity_id filter with wildcards
- support "all" cards
unbenannt

### CODE REVIEW

File: src/panels/lovelace/common/compute-card-element.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  'camera-preview',
+  'entities',
+  'entity-filter',
+  'glance',
+  'history-graph',
+  'media-control',
+  'picture-glance',
+  'plant-status',
+  'weather-forecast'
+];
```
Comment (ciotlosm) [inline src/panels/lovelace/common/compute-card-element.js]: does this need 'error' as well?

Comment (c727) [reply src/panels/lovelace/common/compute-card-element.js]: no, only "good" cards

**Comment by balloob** [comment]
That last commit should not have happened but instead have been a force push ?

File: src/panels/lovelace/common/compute-card-element.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const CARD_TYPES = [
```
Comment (balloob) [inline src/panels/lovelace/common/compute-card-element.js]: Let's not put this method in common. It's only useful for the Lovelace components (`hui-`). So let's keep it in the panel folder. Maybe add a common folder there ?

Comment (c727) [reply src/panels/lovelace/common/compute-card-element.js]: that's what I did ^^

**Comment by c727** [comment]
Forgot to pull after fixing the conclicts in WebUI

File: src/panels/lovelace/common/compute-card-element.js (hunk 2 above)
Comment (balloob) [reply src/panels/lovelace/common/compute-card-element.js]: doh 🤦‍♂️

File: src/panels/lovelace/hui-entity-filter-card.js [hunk 3]
```diff
@@ -1,55 +1,114 @@
...
-    if (filter.domain) {
-      const domain = filter.domain;
-      filters.push(stateObj => computeStateDomain(stateObj) === domain);
+      Object.values(hass.states).forEach((stateObj) => {
+        if (filters.every(filterFunc => filterFunc(stateObj))) {
+          entities.add(stateObj.entity_id);
+        }
+      });
+    });
+    this._cardSize = entities.size;
```
Comment (balloob) [inline src/panels/lovelace/hui-entity-filter-card.js]: We should avoid functions that have side effects. `getCardSize` should just filter and return the length of the array.

Comment (balloob) [reply src/panels/lovelace/hui-entity-filter-card.js]: Wait, it's not even used ?

Comment (c727) [reply src/panels/lovelace/hui-entity-filter-card.js]: Oh yes, I removed it

**Review by balloob — APPROVED** [review]

File: src/panels/lovelace/hui-entity-filter-card.js [hunk 4]
```diff
@@ -1,55 +1,113 @@
...
+    }
+
+    const element = document.createElement(tag);
+    element.config = cardConfig;
+    element.hass = this.hass;
+    this.appendChild(element);
   }
 
-  _computeCardConfig(hass, config) {
-    return Object.assign({}, config.card_config || {}, {
```
Comment (balloob) [inline src/panels/lovelace/hui-entity-filter-card.js]: You broke this. We use `card_config` as the basis for config for next card. We should not copy the filter config. I'll fix this.

**Pull request merged**